### PR TITLE
Jetpack: update Inter font in plans page

### PR DESCRIPTION
**Pull request by keoshi**
## Proposed Changes

* Updates the Inter fonts that were introduced in https://github.com/Automattic/wp-calypso/pull/58671/
* It uses the font Inter Variable, which supports continuous weights of 100 to 900, instead of requesting multiple, discrete weights (Regular, Medium, Semibold, Bold).
* This represents a total savings of ~10% in size, and only 2 HTTP requests (vs 5 before).

Here's the analog diff for jetpack.com: D114865-code

## Testing Instructions

* Fire up this PR.
* Visit the Jetpack pricing page.
* Make sure the variable fonts are being used (e.g.: `Inter-*.var.*`)

### CODE REVIEW

**Review by sgomes — COMMENTED** [review]
Hey @keoshi! 👋

This can definitely turn into an improvement, nice work! 👍 I'll be happy to run some tests for you to see if that's the case.

One thing I noticed, however: is there any particular reason why we're not using `font-display: swap` with the new font, given that we were doing so with the old one? This means we default to `font-display: auto` instead, and if the font arrives late, the page will remain blank for up to 3 seconds. With `display: swap`, the page would be rendered with a fallback font as soon as possible, and then swapped to Inter once that arrives. Given the size of the variable font, there's a fairly good chance it'll arrive late on slower connections.

**Comment by keoshi** [comment]
@sgomes Thanks for the review and apologies for the delay. I was on sabbatical. :)

Updated both the PR and analog diff D114865-code — would appreciate another review so we can get this in.

**Comment by keoshi** [comment]
@sgomes Soft ping on this :)

**Comment by sgomes** [comment]
Sorry for the delay, @keoshi , I haven't had a chance to take a close enough look at this yet. I'll see if I can set aside some time this week 👍

**Comment by keoshi** [comment]
No rush, @sgomes — just wanted to make it didn't go unseen now that I'm back. Thank you in advance!

File: client/my-sites/plans/jetpack-plans/style.scss [hunk 1]
```diff
@@ -4,78 +4,24 @@
...
-	font-display: swap;
-	src:
-		url(https://s0.wp.com/i/fonts/inter/Inter-Medium.woff2?v=3.19) format("woff2"),
-		url(https://s0.wp.com/i/fonts/inter/Inter-Medium.woff?v=3.19) format("woff");
-}
-@font-face {
-	font-family: Inter;
-	font-style: italic;
-	font-weight: 500;
+	font-weight: 100 900;
```
Comment (elliottprogrammer) [inline client/my-sites/plans/jetpack-plans/style.scss:8]: In order for the TeamCity "Code style (Web app)" check to pass, you may need to disable the Stylelint(scales/font-weights) rule for this line:

```suggestion
	/* stylelint-disable-next-line scales/font-weights */
	font-weight: 100 900;
```

File: client/my-sites/plans/jetpack-plans/style.scss [hunk 2]
```diff
@@ -4,78 +4,24 @@
...
-	src:
-		url(https://s0.wp.com/i/fonts/inter/Inter-SemiBold.woff2?v=3.19) format("woff2"),
-		url(https://s0.wp.com/i/fonts/inter/Inter-SemiBold.woff?v=3.19) format("woff");
-}
-@font-face {
-	font-family: Inter;
-	font-style: italic;
-	font-weight: 600;
-	font-display: swap;
+	font-named-instance: "Regular";
```
Comment (elliottprogrammer) [inline client/my-sites/plans/jetpack-plans/style.scss:12]: And this line is throwing a stylelint error too, causing the "Code style (Web app)" check to fail. You could disable the rule here too:

```suggestion
	/* stylelint-disable-next-line property-no-unknown */
	font-named-instance: "Regular";
```

**Review by elliottprogrammer — APPROVED** [review]
Tested and works good, as described. The _variable_ Inter font is being used. 👍
LGTM! 🙌

File: client/my-sites/plans/jetpack-plans/style.scss (hunk 1 above)
Comment (keoshi) [reply client/my-sites/plans/jetpack-plans/style.scss:8]: Thanks, @elliottprogrammer — I've updated both instances where this exists.

File: client/my-sites/plans/jetpack-plans/style.scss (hunk 2 above)
Comment (keoshi) [reply client/my-sites/plans/jetpack-plans/style.scss:12]: Also added comments to both of these properties.

**Comment by keoshi** [comment]
Thanks so much for the review, @elliottprogrammer! Appreciate you.

I've updated all instances where you were seeing the stylelint issues. Hopefully it's all good now.

**Review by elliottprogrammer — APPROVED** [review]
LGTM! 👍
Hmm, except that more checks are failing now.. You probably just need to update your branch with trunk (do a rebase). I bet that gets everything passing. 😉

**Review by sgomes — APPROVED** [review]
Sorry for taking such a long time to get to this, @keoshi , I had some AFK time slip in between as well 😞

The code changes look good, so if you're only looking for a code-level approval please feel free to go right ahead and commit. I'll be doing some performance testing based on D114865-code and posting the numbers here, in any case (which may be insufficient for proper testing, but may be enough to give us an idea of impact), but there's no need tp wait for that.

**Review by sgomes — APPROVED** [review]
I was unable to test the wpcom patch, as it couldn't be applied to my sandbox (there may have been some code restructuring since it was created).

That said, looking at the code again, I'm confident this change can go ahead without further testing, as preserving `font-display: swap` should ensure a quick first render, even if the new font happens to take a bit longer to load.

**Comment by keoshi** [comment]
@elliottprogrammer Could you help get this merged, please? I'm trying to rerun the tests in TC, but getting a `Build with id 'X' not found`. Thanks in advance!

**Comment by elliottprogrammer** [comment]
> @elliottprogrammer Could you help get this merged, please? I'm trying to rerun the tests in TC, but getting a `Build with id 'X' not found`. Thanks in advance!

@keoshi, Yes sure! I rebased the branch and then all tests were passing! ☑ 😄
So I went ahead and deployed it. 👍 🙌

**Comment by keoshi** [comment]
Thank you so much, @elliottprogrammer !!

**Pull request merged**